src/firefox/api.js

```diff
@@ -0,0 +1,194 @@
+"use strict";
+
+const lazy = {};
+
+XPCOMUtils.defineLazyServiceGetters(lazy, {
+  gMIMEService: ["@mozilla.org/mime;1", "nsIMIMEService"],
+});
+
+const https = "https";
+const iconSize = 32;
+const browserNamesWin = ["Chrome", "Edge", "Opera", "Safari"];
+const browserNamesMac = ["Safari", "Chrome", "Microsoft Edge", "Opera", "Arc"];
+let logs = [];
+
+this.experiments_firefox_launch = class extends ExtensionAPI {
+  getAPI(context) {
+    return {
+      experiments: {
+        firefox_launch: {
+          _isValidHandlerExecutable(aExecutable) {
+            let leafName;
+            if (AppConstants.platform == "win") {
+              leafName = `${AppConstants.MOZ_APP_NAME}.exe`;
+            } else if (AppConstants.platform == "macosx") {
+              leafName = AppConstants.MOZ_MACBUNDLE_NAME;
+            } else {
+              return false;
+            }
+            return (
+              aExecutable &&
+              aExecutable.exists() &&
+              aExecutable.isExecutable() &&
+              aExecutable.leafName !== leafName
+            );
+          },
+
+          _isValidHandlerApp(aHandlerApp) {
+            if (!aHandlerApp) {
+              return false;
+            }
+
+            if (aHandlerApp instanceof Ci.nsILocalHandlerApp) {
+              return this._isValidHandlerExecutable(aHandlerApp.executable);
+            }
+            return false;
+          },
+
+          // Get all windows able to open https protocol
+          _getAvailableBrowsersWin() {
+            let mimeInfo = lazy.gMIMEService.getFromTypeAndExtension(
+              "text/html",
+              "html"
+            );
+            let appList = mimeInfo.possibleLocalHandlers || [];
+            let appDataList = [];
+            for (let idx = 0; idx < appList.length; idx++) {
+              let app = appList.queryElementAt(idx, Ci.nsILocalHandlerApp);
+              logs.push("App: " + app.executable.path);
+              if (!this._isValidHandlerApp(app)) {
+                logs.push("Invalid app");
```

Comment:
I was collecting logs because console logs did not show up in the background script here, so I would return them. I have recently setup extension development with the firefox source code so I no longer need these log collections.

Reply:
PR: https://github.com/mozilla/firefox-launch/pull/39